django/dispatch/dispatcher.py

```diff
@@ -410,7 +423,10 @@ def _clear_dead_receivers(self):
             self.receivers = [
                 r
                 for r in self.receivers
-                if not (isinstance(r[1], weakref.ReferenceType) and r[1]() is None)
+                if (
+                    not (isinstance(r[1], weakref.ReferenceType) and r[1]() is None)
+                    and not (isinstance(r[2], weakref.ReferenceType) and r[2]() is None)
```

Comment:
trivia: now that sender_ref can only be a weakref or None, it might be clearer to check `is not None` rather than `isinstance`? This differs from the receiver_ref which can be strong.

Reply:
yeah I think it's worth tweaking, will do.